Libraries/Utilities/MessageQueue.js

```diff
@@ -182,18 +188,22 @@ class MessageQueue {
       this._lastFlush = now;
     }
     Systrace.counterEvent('pending_js_to_native_queue', this._queue[0].length);
-    if (__DEV__ && SPY_MODE && isFinite(module)) {
-      console.log('JS->N : ' + this._remoteModuleTable[module] + '.' +
```

Comment:
Yes!
I'm thinking perhaps a default 'spy' implementation, that you can use and is built-in.
Something like [this](https://github.com/jondot/rn-snoopy/blob/master/log.js) and [this](https://github.com/jondot/rn-snoopy/blob/master/formatting.js).


The ergonomics would look like this:

```javascript
MessageQueue.spy()
```
sets the default logging strategy in place.

Then clearing a spy, disables tracing, and would have to be done like so

```javascript
MessageQueue.spy(null)
```

WDYT?